cmd/jaeger/internal/extension/jaegerquery/server.go

```diff
@@ -140,23 +140,41 @@ func (s *server) addArchiveStorage(
 		return nil
 	}
 
-	f, err := jaegerstorage.GetStorageFactory(s.config.Storage.TracesArchive, host)
+	f, err := jaegerstorage.GetTraceStoreFactory(s.config.Storage.TracesArchive, host)
 	if err != nil {
-		return fmt.Errorf("cannot find archive storage factory: %w", err)
+		return fmt.Errorf("cannot find traces archive storage factory: %w", err)
 	}
 
-	if !opts.InitArchiveStorage(f, s.telset.Logger) {
-		s.telset.Logger.Info("Archive storage not initialized")
+	reader, err := f.CreateTraceReader()
+	if err != nil {
+		s.telset.Logger.Error("Cannot init traces archive storage reader", zap.Error(err))
+		return nil
+	}
+	writer, err := f.CreateTraceWriter()
+	if err != nil {
+		s.telset.Logger.Error("Cannot init traces archive storage writer", zap.Error(err))
+		return nil
 	}
+	v2opts.ArchiveTraceReader = reader
+	v2opts.ArchiveTraceWriter = writer
 
-	ar, aw := v1adapter.InitializeArchiveStorage(f, s.telset.Logger)
-	if ar != nil && aw != nil {
-		v2opts.ArchiveTraceReader = ar
-		v2opts.ArchiveTraceWriter = aw
-	} else {
-		s.telset.Logger.Info("Archive storage not initialized")
+	v1Reader, err := v1adapter.GetV1Reader(reader)
+	if err != nil {
+		// if the spanstore.Reader is not available, downgrade the native tracestore.Reader to
+		// a spanstore.Reader
+		v1Reader = v1adapter.NewSpanReader(reader)
```

Comment:
@yurishkuro this branch is not reachable with the current setup in the tests because `jaegerstorage.GetTraceStoreFactory` always return a v1 factory wrapped in an adapter. As a result, we never have a case where the v1 reader cannot be obtained. Thoughts on how to proceed?